lib/ogr/coordinate_transformation.rb

```diff
@@ -1,44 +1,30 @@
 # frozen_string_literal: true
 
+require 'ffi'
 require 'narray'
 require_relative '../ogr'
 require_relative '../gdal'
 
 module OGR
   class CoordinateTransformation
-    # @param proj4_source [String]
-    # @return [String]
-    def self.proj4_normalize(proj4_source)
-      if GDAL._supported?(:OCTProj4Normalize)
-        FFI::GDAL::GDAL.OCTProj4Normalize(proj4_source)
-      else
-        raise OGR::UnsupportedOperation,
-              'Your version of GDAL/OGR does not support OCTProj4Normalize'
-      end
-    end
+    class AutoPointer < ::FFI::AutoPointer
+      def self.release(c_pointer)
+        return if c_pointer.nil? || c_pointer.null?
 
-    # @param pointer [FFI::Pointer]
-    def self.release(pointer)
-      return unless pointer && !pointer.null?
-
-      FFI::OGR::SRSAPI.OCTDestroyCoordinateTransformation(pointer)
+        FFI::OGR::SRSAPI.OCTDestroyCoordinateTransformation(c_pointer)
+      end
     end
 
-    # @return [OGR::SpatialReference]
-    attr_reader :source_coordinate_system
-
-    # @return [OGR::SpatialReference]
-    attr_reader :destination_coordinate_system
-
-    # @return [FFI::Pointer] C pointer that represents the CoordinateTransformation.
+    # @return [OGR::CoordinateTransformation::AutoPointer] C pointer that
+    #   represents the CoordinateTransformation.
     attr_reader :c_pointer
 
     # @param source_srs [OGR::SpatialReference]
     # @param destination_srs [OGR::SpatialReference]
     # @raise [FFI::GDAL::InvalidPointer]
     def initialize(source_srs, destination_srs)
-      source_ptr = GDAL._pointer(source_srs)
-      destination_ptr = GDAL._pointer(destination_srs)
```

Comment:
FWIW, I feel like this became an anti-pattern, obfuscating what's going on in that method when we really should be able to determine in each method whether we're working with a borrowed or owned pointer. ...thus I'm getting rid of these calls as I go, where it makes sense.